### PR TITLE
Prebid Auctions

### PR DESCRIPTION
**Pull request by protonate**
Prebid Auctions are implemented which enables instances of Prebid requests, responses and ad renders. This PR also includes the API changes needed to make concurrent requests for multiple placements, to queue up the next set of placements (for example, in an image carousel app) and to connect bid requests with bid responses across auction instances for greater level of analytics details.

### CODE REVIEW

**Comment by protonate** [comment]
@mkendall07 for discussion

**Comment by nedstankus** [comment]
What's the story with the "on hold" label?

**Comment by mkendall07** [comment]
@nedstankus 
We are too close to a release to risk adding more regression with this PR.

File: src/adapters/pubmatic.js [hunk 1]
```diff
@@ -73,57 +73,59 @@ var PubmaticAdapter = function PubmaticAdapter() {
...
+          adResponse.ad = unescape(adUnit.creative_tag);  // jshint ignore:line
+          adResponse.ad += utils.createTrackPixelHtml(decodeURIComponent(adUnit.tracking_url));
+          adResponse.width = dimensions[0];
+          adResponse.height = dimensions[1];
+          adResponse.dealId = adUnitInfo.wdeal;
+
+          bidmanager.addBidResponse(bids[i].placementCode, adResponse);
+        } else {
+          // Indicate an ad was not returned
+          adResponse = bidfactory.createBid(2);
```
Comment (protonate) [inline src/adapters/pubmatic.js]: Need to pass bid ID for requested bid here.

**Comment by protonate** [comment]
We are going to close this PR and handling the concurrent bid request issue by queuing requests rather than creating auction instances. This keeps the singleton architecture of Prebid.js, at least for now, and we will work with each bidder team towards the changes needed within each adapter to achieve auction instances if that is defined as a roadmap priority by the Prebid Core Team. Issues #333 and #360 will be accommodated by the bid request queue approach.

**Pull request closed without merging**